### ISSUE TITLE
Make inheritance of original `impl` and method trait bounds opt-in

### ISSUE THREAD

**Issue by bonsairobo** [bonsairobo/rkyv_impl#3]
When trait bounds are always inherited, this may over-constrain the generated methods, causing superfluous bounds to propagate through all generic code.

We could have an opt-in syntax like:

```rust
#[archive_impl(inherit, bounds(...))]
impl<T> Foo<T> {
    #[archive_method(inherit, bounds(...))]
    fn method(...) { ... }
}
```

**Comment by bonsairobo** [comment]
We could even get more specific with this, and do `inherit(T, S)`, meaning that only the bounds from those type params are inherited. We could also do something like `archived_inherit(T)`, meaning that any `T` bounds would be transformed into `T::Archived` bounds; that might get a little too tricky though to make it work consistently in all cases.

**Comment by bonsairobo** [comment]
Fixed by cf96646
